.github/workflows/check-network.yaml

```diff
@@ -59,53 +67,92 @@ jobs:
       - name: Terraform Init
         run: terraform init
 
-      - name: Prepare Environment
-        env:
-          GITHUB_SHA: ${{ github.sha }}
-          GITHUB_REF: ${{ github.ref }}
-        run: ./scripts/prepare-env.sh
-
-      - name: Terraform Apply
-        run: |
-          export TF_VAR_cluster_name=$(cat CLUSTER_NAME)
-
-          # Start SSH Agent
-          eval `ssh-agent -s`
-
-          terraform apply -auto-approve
-
-      - name: Bindata
+      - name: Cache embedded binaries
         uses: actions/cache@v3
-        id: generated-bindata
         with:
+          key: ${{ runner.os }}-embedded-bins-linux-${{ hashFiles('embedded-bins/**/*') }}
```

Comment:
Not sure if we are okay with this cross-workflow cache sharing. If yes, we need to keep in mind that stuff built by the usual branch builds may end up here, and vice versa stuff that's been built here will potentially end up in branch builds. Should be _relatively_ safe, though, since we're hashing across everything below embedded-bins and there's no crucial variable overriding going on, neither in the Makefile nor in the workflows.